notifications/pipeline_steps/base.py

```diff
@@ -24,5 +24,7 @@ def process_message(message):
         logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
         if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
```

Comment:
The expiration time check would happen microseconds before calling the pipeline step's `process_message` method - since it would be called from within the `Pipeline.process` method _after_ the message is received by the queue/process.  So there is no lag in time that we need to worry about.